Tests/Runtime/BrainTargetOverrideTests.cs

```diff
@@ -0,0 +1,136 @@
+                Assert.That(m_CameraHolderWithBrain.GetComponent<Camera>().fieldOfView == m_CameraHolderWithoutBrain.GetComponent<Camera>().fieldOfView, Is.True);
+                Assert.That(EqualTransforms(m_CameraHolderWithBrain.transform, m_CameraHolderWithoutBrain.transform), Is.True);
+                Assert.That(EqualTransforms(m_CameraHolderWithBrain.transform, m_GoWithBrain.transform), Is.True);
+                Assert.That(EqualTransforms(m_CameraHolderWithBrain.transform, m_GoWithoutBrain.transform), Is.True);
+            
+                bool EqualTransforms(Transform a, Transform b) => a.position == b.position && a.rotation == b.rotation;
+            }
+        }
+        
+        IEnumerator CheckDisconnectedBrains()
+        {
+            m_BrainAlone.ControlledObject = null;
+            m_BrainAlone2.ControlledObject = null;
+            m_FollowObject.transform.position += m_Delta;
+            yield return null;
+            var position = m_CameraHolderWithBrain.transform.position;
+            Assert.That(position == m_CameraHolderWithoutBrain.transform.position, Is.False);
+            Assert.That(position == m_GoWithoutBrain.transform.position, Is.False);
+        }
+        
+        [UnityTest]
+        public IEnumerator DoNothing()
+        {
+            m_Vcam.m_Lens.FieldOfView = 50;
+            yield return CheckThatBrainsAreControllingTheirTargets();
+        }
+
+        [UnityTest]
+        public IEnumerator ThirdPerson()
+        {
+            m_Vcam.AddCinemachineComponent<Cinemachine3rdPersonFollow>();
+            m_Vcam.m_Lens.FieldOfView = 50;
+            m_Vcam.Follow = m_FollowObject.transform;
+            yield return CheckThatBrainsAreControllingTheirTargets();
+            yield return CheckDisconnectedBrains();
+        }
+        
+        [UnityTest]
+        public IEnumerator FramingTransposer()
+        {
+            var component = m_Vcam.AddCinemachineComponent<CinemachineFramingTransposer>();
+            component.m_XDamping = 0;
+            component.m_YDamping = 0;
+            component.m_ZDamping = 0;
+            component.m_CameraDistance = 1f;
+            m_Vcam.Follow = m_FollowObject.transform;
+            yield return CheckThatBrainsAreControllingTheirTargets();
+            yield return CheckDisconnectedBrains();
+        }
+
+        [UnityTest]
+        public IEnumerator HardLockToTarget()
+        {
+            m_Vcam.AddCinemachineComponent<CinemachineHardLockToTarget>();
+            m_Vcam.Follow = m_FollowObject.transform;
+            yield return CheckThatBrainsAreControllingTheirTargets();
+            yield return CheckDisconnectedBrains();
+        }
+
+        [UnityTest]
+        public IEnumerator OrbTransposer()
+        {
+            var component = m_Vcam.AddCinemachineComponent<CinemachineOrbitalTransposer>();
+            component.m_XAxis.m_InputAxisName = ""; // to avoid error when the Input System is installed in the testing project
```

Comment:
This line is not needed after https://github.com/Unity-Technologies/com.unity.cinemachine/pull/347 lands.